application/src/main/java/org/thingsboard/server/controller/DeviceController.java

```diff
@@ -141,6 +141,45 @@ public Device saveDevice(@RequestBody Device device,
         }
     }
 
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/bulk/devices", method = RequestMethod.POST)
+    @ResponseBody
+    public List<Device> saveBulkDevices(@RequestBody List<Device> devices,
+                                        @RequestParam(name = "accessToken", required = false) String accessToken) throws ThingsboardException {
+        List<Device> savedDevices = new ArrayList<>();
+        for (Device device : devices) {
+            try {
+                device.setTenantId(getCurrentUser().getTenantId());
+
+                checkEntity(device.getId(), device, Resource.DEVICE);
+
+                Device savedDevice = checkNotNull(deviceService.saveDeviceWithAccessToken(device, accessToken));
```

Comment:
This is nonsence. There is a unique key for access token. You can't save multiple devices with the same token.